lib/Runtime/Types/PathTypeHandler.cpp

```diff
@@ -1521,6 +1521,7 @@ namespace Js
         DynamicType* cachedDynamicType = nullptr;
         BOOL isJsrtExternalType = instance->GetType()->IsJsrtExternal();
         DynamicType* oldType = isJsrtExternalType ? 0 : instance->GetDynamicType();
```

Comment:
`oldType` will not get collected because `instance` is holding on to it. We replace it with `cachedDynamicType` at the end of this method. But as I mentioned in the other comment, `oldType` can get reclaimed and re-allocated after this function in which case we will get wrong `cachedDynamicType`.